content_restrictions/content_restrictions.py

```diff
@@ -168,11 +211,64 @@ def has_access_to_content(self):
         Check if the user has access to the content.
 
         - If password restriction is enabled, check if the user has entered the correct password.
+        - If IP restriction is enabled, check if the user has access with the IP whitelist.
         """
-        if self.password_restriction:
-            return bool(self.user_provided_password == self.password)
+        if self.password_restriction and not self.is_correct_password:
+            return False
+        if self.ip_restriction:
+            return self.has_access_with_ip_whitelist(get_current_request())
         return True
 
+    @property
+    def is_correct_password(self) -> bool:
+        """
+        Check if the user has entered the correct password.
+
+        Returns:
+            bool: True if the user has entered the correct password, False otherwise.
+        """
+        return self.user_provided_password == self.password
+
+    def has_access_with_ip_whitelist(self, request) -> bool:
+        """
+        Check if the user has access to the content based on the IP whitelist.
+
+        Args:
+            request (WSGIRequest): The current request object.
+
+        Returns:
+            bool: True if the user has access to the content, False otherwise.
+        """
+        client_ips = ip.get_all_client_ips(request)
+        ip_ranges = self.ip_ranges_whitelist.split(",")
+        for ip_range in ip_ranges:
+            start_ip, end_ip = ip_range.split("-")
+            for client_ip in client_ips:
+                if self.ip_in_whitelist(client_ip, start_ip, end_ip):
+                    return True
+        return False
+
+    @staticmethod
+    def ip_in_whitelist(ip_address: str, start_ip: str, end_ip: str) -> bool:
+        """
+        Check if the IP is in the whitelist.
+
+        Args:
+            ip_address (str): The IP address to check.
+            start_ip (str): The start of the IP range.
+            end_ip (str): The end of the IP range.
+
+        Returns:
+            bool: True if the IP is in the whitelist, False otherwise.
+        """
+        try:
+            ip_address = ipaddress.IPv4Address(ip_address)
+            start_ip = ipaddress.IPv4Address(start_ip)
+            end_ip = ipaddress.IPv4Address(end_ip)
```

Comment:
so IPV6 is not supported?

Reply:
We could use `ipaddress.ip_address()` instead, to accept IPv6.

Reply:
Added IPv6 support.